artpaint/application/PaintApplication.cpp

```diff
@@ -476,6 +479,64 @@ PaintApplication::SetColor(rgb_color color, bool foreground)
 }
 
 
+void
+PaintApplication::_InstallMimeTypes()
+{
+	// install mime type of documents
+	BMimeType mime(HS_PROJECT_MIME_STRING);
+	if (mime.InitCheck() < B_OK)
+		return;
+
+	BString snifferRule;
+	if (mime.IsInstalled() && mime.GetSnifferRule(&snifferRule)
+			== B_OK && snifferRule.Length() > 0)
+		return;
+
+	mime.Delete();
+
+	status_t ret = mime.Install();
+	if (ret < B_OK) {
+		fprintf(stderr, "Could not install mime type '" HS_PROJECT_MIME_STRING
+			"': %s.\n", strerror(ret));
+		return;
+	}
+
+	// set preferred app
+	if (mime.SetPreferredApp("application/x-vnd.artpaint") < B_OK)
+		fprintf(stderr, "Could not set preferred app!\n");
+
+	// set descriptions
+	if (mime.SetShortDescription("ArtPaint project") < B_OK)
+		fprintf(stderr, "Could not set short description of mime type!\n");
+	if (mime.SetLongDescription("ArtPaint project format containing layers etc.") != B_OK) {
+		fprintf(stderr, "Could not set long description of mime type!\n");
+	}
+
+	// set sniffer rule
+	snifferRule = "0.9 (0xffffffff010101010200000002020202001100110800000000000000)";
```

Comment:
is this for sure correct?  Based on my experience with file/libmagic, most signatures weren't quite that long.